packages/core/frontend/sirius-components-core/src/workbench/Workbench.types.ts

```diff
@@ -30,6 +30,7 @@ export type RepresentationMetadata = {
   id: string;
   label: string;
   kind: string;
+  iconURLs?: string[];
```

Comment:
Why would this be optional? It is explicitly defined as non null in the GraphQL API `iconURLs: [String!]!`

Reply:
Every time someone uses something like `?` or `null` or `undefined` in an odd location, it should SCREAM to everyone involved that this is not working as expected. The only questions are where is the issue located and why am I the only one bothered by this.

Of course, now that I've looked it for a couple of minutes, the new field `RepresentationMetadata#iconURLs` only works if the representation metadata are requested by one very specific path in the GraphQL API, the only path tested I suppose. If you ask for something like this:


```
viewer {
  editingContext {
    representation {
      iconURLs
    }
  }
}
```

Then the backend returns null which is not a valid value because `EditingContextRepresentationDataFetcher` does not support it because it returns a `RepresentationMetadata` instead of `RepresentationMetadataDTO`.

This work has not been completed.